src/dom.ts

```diff
@@ -200,29 +200,56 @@ export function containsClass(element: HTMLElement, className: string): boolean
  * @example
  * var fragment = dom.fromString('<tr>');
  */
-export function fromString(html: string): DocumentFragment {
-	html = String(html);
+export function fromString(html: string): DocumentFragment  {
+	let fragment: DocumentFragment;
+	let useContextualFragment = has('dom-contextual-fragment');
 
-	const match = html.match(/<\s*([\w\:]+)/);
-	const tag = match ? match[1].toLowerCase() : '';
 	let master: HTMLElement = document.createElement('div');
```

Comment:
It does assign the correct type (HTMLDivElement), but the unwrapping code around line 248 requires the more generic HTMLElement type since we're reassigning the master element as it drills down through the wrapping elements. I'll look at this when I look at the unwrapping code.